src/components/MyTemplate/MyTemplateContainer.tsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable require-jsdoc */
+import React from "react";
+import MyTemplateListItem from "./MyTemplateListItem";
+import { Flex, Text, Button } from "@chakra-ui/react";
+import { IMyTemplates } from "@/interfaces/template";
+import { useDispatch } from "react-redux";
+import { switchMyTemplateDisplay } from "@/store/reducer/buttonToggleSlice";
+
+interface myTemplates {
+  myTemplates: IMyTemplates[] | undefined;
+}
+
+export default function MyTemplateContainer({ myTemplates }: myTemplates) {
+  const disPatch = useDispatch();
+  const handleReturnToDesign = () => {
+    disPatch(switchMyTemplateDisplay(false));
+  };
+  const myTemplateLists = (): JSX.Element[] | undefined => {
+    return myTemplates
+      ?.sort((itemA, itemB) => +itemB.createdAt - +itemA.createdAt)
+      .map((item) => <MyTemplateListItem key={item._id} {...item} />);
```

Comment:
sort is invalid, because _createdAt_ is not a pure number, so NaN is converted. In addition, the data sent from the backend has been sorted by _createdAt_, and there is no need to sort it again.

Reply:
> you do not need to sort the templates by date in frontend, they have already been sorted in backend

fixed thanks all